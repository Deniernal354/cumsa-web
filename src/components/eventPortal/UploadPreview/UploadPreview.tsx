"use client";
import { useState, ChangeEvent } from "react";
import { EventPreview } from "../EventPreview";

export function UploadPreview({ images }: { images: string[] }) {
  const [selectedImage, setSelectedImage] = useState<string | undefined>(undefined);
  const [imageStyle, setImageStyle] = useState<
    "cover" | "contain" | "fill" | "scale-down" | "none"
  >("cover");

  return (
    <div className="flex flex-col md:flex-row items-start gap-8 justify-center">
      <div className="flex flex-col gap-2">
        <span className="text-sm font-medium text-gray-600">Preview Selected Image</span>
        <EventPreview
          event={{
            title: "Sample Event",
            date: "2024-01-01",
            location: "Sample Location",
            image: selectedImage,
            imageStyle: imageStyle,
          }}
        />
      </div>
      <div className="flex flex-col gap-4 max-w-xs">
        <div className="flex flex-col gap-2">
          <span className="text-sm font-medium text-gray-600">Available Images</span>
          <select
            className="w-full border rounded px-3 py-2 text-sm"
            value={selectedImage || ""}
            onChange={(e) => setSelectedImage(e.target.value || undefined)}
          >
            <option value="">Select an image to preview</option>
            {images.map((img) => (
              <option key={img} value={img}>
                {img}
              </option>
            ))}
          </select>
        </div>
        <div className="flex flex-col gap-2">
          <span className="text-sm font-medium text-gray-600">Image Display Style</span>
          <select
            className="w-full border rounded px-3 py-2 text-sm"
            value={imageStyle}
            onChange={(e: ChangeEvent<HTMLSelectElement>) =>
              setImageStyle(e.target.value as "cover" | "contain" | "fill" | "scale-down" | "none")
            }
          >
            <option value="cover">Cover (crop to fill)</option>
            <option value="contain">Contain (fit entire image)</option>
            <option value="fill">Fill (stretch to fit)</option>
            <option value="scale-down">Scale Down (shrink if needed)</option>
            <option value="none">None (original size)</option>
          </select>
        </div>
      </div>
    </div>
  );
}
