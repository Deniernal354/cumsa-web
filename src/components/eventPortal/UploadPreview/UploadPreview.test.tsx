import { render } from "@testing-library/react";
import { UploadPreview } from "./UploadPreview";

test("UploadPreview snapshot", () => {
  const images = ["/events/test1.jpg", "/events/test2.png"];
  const { asFragment } = render(<UploadPreview images={images} />);
  expect(asFragment()).toMatchSnapshot();
});
