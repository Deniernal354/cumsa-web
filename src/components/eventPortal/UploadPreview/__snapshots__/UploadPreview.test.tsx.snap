// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadPreview snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col md:flex-row items-start gap-8 justify-center animate-fade-in"
  >
    <div
      class="flex flex-col gap-2"
    >
      <span
        class="text-sm font-medium text-[var(--navy)]"
      >
        Preview Selected Image
      </span>
      <div
        class="group relative bg-white rounded-2xl shadow-lg hover:shadow-xl transition-all duration-300 overflow-hidden transform hover:-translate-y-1 w-60"
      >
        <div
          class="relative aspect-[4/3] overflow-hidden"
        >
          <div
            class="w-full h-full flex items-center justify-center bg-gradient-to-br from-gray-50 to-gray-100 border-2 border-dashed border-gray-300"
          >
            <div
              class="text-center p-4"
            >
              <svg
                class="w-8 h-8 text-gray-400 mx-auto mb-2"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 64H96a64.07 64.07 0 0 0-64 64v256a64.07 64.07 0 0 0 64 64h320a64.07 64.07 0 0 0 64-64V128a64.07 64.07 0 0 0-64-64zm-80 64a48 48 0 1 1-48 48 48.05 48.05 0 0 1 48-48zM96 416a32 32 0 0 1-32-32v-67.63l94.84-84.3a48.06 48.06 0 0 1 65.8 1.9l64.95 64.81L172.37 416zm352-32a32 32 0 0 1-32 32H217.63l121.42-121.42a47.72 47.72 0 0 1 61.64-.16L448 333.84z"
                />
              </svg>
              <p
                class="text-xs text-gray-500"
              >
                No image selected
              </p>
            </div>
          </div>
          <div
            class="absolute inset-0 bg-gradient-to-t from-black/70 via-black/20 to-transparent opacity-0 group-hover:opacity-100 transition-opacity duration-300"
          />
          <div
            class="absolute top-2 left-2 px-2 py-1 bg-blue-600 text-white text-xs font-medium rounded-full shadow-lg"
          >
            Preview
          </div>
        </div>
        <div
          class="p-4 space-y-2"
        >
          <h3
            class="font-bold text-gray-900 text-sm leading-tight group-hover:text-blue-600 transition-colors duration-200 overflow-hidden"
            style="display: -webkit-box; -webkit-line-clamp: 2;"
          >
            Sample Event
          </h3>
          <div
            class="space-y-1.5 text-xs text-gray-600"
          >
            <div
              class="flex items-center gap-2"
            >
              <svg
                class="w-3 h-3 text-blue-500 flex-shrink-0"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="none"
                  height="384"
                  rx="48"
                  stroke-linejoin="round"
                  stroke-width="32"
                  width="416"
                  x="48"
                  y="80"
                />
                <circle
                  cx="296"
                  cy="232"
                  r="24"
                />
                <circle
                  cx="376"
                  cy="232"
                  r="24"
                />
                <circle
                  cx="296"
                  cy="312"
                  r="24"
                />
                <circle
                  cx="376"
                  cy="312"
                  r="24"
                />
                <circle
                  cx="136"
                  cy="312"
                  r="24"
                />
                <circle
                  cx="216"
                  cy="312"
                  r="24"
                />
                <circle
                  cx="136"
                  cy="392"
                  r="24"
                />
                <circle
                  cx="216"
                  cy="392"
                  r="24"
                />
                <circle
                  cx="296"
                  cy="392"
                  r="24"
                />
                <path
                  d="M128 48v32m256-32v32"
                  fill="none"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="32"
                />
                <path
                  d="M464 160H48"
                  fill="none"
                  stroke-linejoin="round"
                  stroke-width="32"
                />
              </svg>
              <span
                class="font-medium"
              >
                Jan 1, 2024
              </span>
            </div>
            <div
              class="flex items-center gap-2"
            >
              <svg
                class="w-3 h-3 text-green-500 flex-shrink-0"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 32C167.67 32 96 96.51 96 176c0 128 160 304 160 304s160-176 160-304c0-79.49-71.67-144-160-144zm0 224a64 64 0 1 1 64-64 64.07 64.07 0 0 1-64 64z"
                />
              </svg>
              <span
                class="truncate"
              >
                Sample Location
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col gap-4 max-w-xs"
    >
      <div
        class="flex flex-col gap-2"
      >
        <span
          class="text-sm font-medium text-[var(--navy)]"
        >
          Available Images
        </span>
        <select
          class="w-full border-2 border-[color:var(--navy)]/20 rounded-lg px-3 py-2 text-sm focus:border-[var(--blue)] focus:outline-none"
        >
          <option
            value=""
          >
            Select an image to preview
          </option>
          <option
            value="/events/test1.jpg"
          >
            /events/test1.jpg
          </option>
          <option
            value="/events/test2.png"
          >
            /events/test2.png
          </option>
        </select>
      </div>
      <div
        class="flex flex-col gap-2"
      >
        <span
          class="text-sm font-medium text-[var(--navy)]"
        >
          Image Display Style
        </span>
        <select
          class="w-full border-2 border-[color:var(--navy)]/20 rounded-lg px-3 py-2 text-sm focus:border-[var(--blue)] focus:outline-none"
        >
          <option
            value="cover"
          >
            Cover (crop to fill)
          </option>
          <option
            value="contain"
          >
            Contain (fit entire image)
          </option>
          <option
            value="fill"
          >
            Fill (stretch to fit)
          </option>
          <option
            value="scale-down"
          >
            Scale Down (shrink if needed)
          </option>
          <option
            value="none"
          >
            None (original size)
          </option>
        </select>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
